python/tests/scripts/test_update_storage.py

```diff
@@ -37,139 +39,97 @@ def _get_azure_storage_config(cfg):
-    assert s3_storage.credential_key == ""
+    storage_config = _get_s3_storage_config(config)
+    assert storage_config.bucket_name == ""
+    assert storage_config.credential_name == ""
+    assert storage_config.credential_key == ""
 
 
-def get_s3_aws_auth_uri_bucket_and_setup_server(moto_s3_endpoint_and_credentials, monkeypatch):
-    endpoint, port, bucket, aws_access_key, aws_secret_key = moto_s3_endpoint_and_credentials
-    monkeypatch.setenv("AWS_ACCESS_KEY_ID", aws_access_key)
-    monkeypatch.setenv("AWS_SECRET_ACCESS_KEY", aws_secret_key)
-    return (
-        endpoint.replace("http://", "s3://").rsplit(":", 1)[0] + ":" + bucket + "?aws_auth=true" + "&port=" + port,
-        bucket,
-    )
+def test_upgrade_script_s3_rbac_ok(s3_storage: S3Bucket, monkeypatch):
+    """Just _RBAC_ as creds is a placeholder. Leave config with that alone."""
+    if os.name == "nt":
+        if sys.version_info < (3, 9):
+            pytest.skip("Older Python don't support unsetenv on Windows")
 
+        # We statically linked msvcrt which maintains a seprate copy of the environment
+        from arcticdb_ext.tools import putenv_s
 
-@pytest.mark.skipif(sys.platform == "win32", reason="Test fixture issue with creds")
-def test_upgrade_script_s3_rbac_ok(moto_s3_endpoint_and_credentials, monkeypatch):
-    """Just _RBAC_ as creds is a placeholder. Leave config with that alone."""
-    uri, bucket = get_s3_aws_auth_uri_bucket_and_setup_server(moto_s3_endpoint_and_credentials, monkeypatch)
+        monkeypatch.setattr(os, "putenv", putenv_s)
+        monkeypatch.setattr(os, "unsetenv", lambda n: putenv_s(n, ""))
 
-    ac = Arctic(uri)
-    create_library_config(ac, LIB_NAME)
+    monkeypatch.setenv("AWS_ACCESS_KEY_ID", s3_storage.key.id)
+    monkeypatch.setenv("AWS_SECRET_ACCESS_KEY", s3_storage.key.secret)
+
+    uri = s3_storage.replace_uri_field(s3_storage.arctic_uri, ArcticUriFields.USER, "aws_auth=true", start=1)
+    uri = s3_storage.replace_uri_field(uri, ArcticUriFields.PASSWORD, "", start=1, end=3)
+    s3_storage.arctic_uri = uri
+
+    ac = s3_storage.create_arctic()  # create_arctic() does proper clean up
+    create_library_config(ac, LIB_NAME)  # Note this avoids the override so _RBAC_ is actually written to storage
+    ac[LIB_NAME].write_pickle("x", 123)
 
     run(uri=uri, run=True)
 
+    ac = Arctic(uri)
     config = ac._library_manager.get_library_config(LIB_NAME)
     s3_storage = _get_s3_storage_config(config)
-    assert s3_storage.bucket_name == bucket
+    assert s3_storage.bucket_name == s3_storage.bucket_name
     assert s3_storage.credential_name == USE_AWS_CRED_PROVIDERS_TOKEN
     assert s3_storage.credential_key == USE_AWS_CRED_PROVIDERS_TOKEN
 
+    lib = ac[LIB_NAME]
+    assert lib.read("x").data == 123
 
-@pytest.mark.skipif(sys.platform == "win32", reason="Test fixture issue with creds")
-def test_aws_auth_with_storage_overrider(moto_s3_endpoint_and_credentials, monkeypatch):
```

Comment:
This test is a bit messy, but it seems to be our only check with `aws_auth=True`, so I don't think we should delete it completely.

Reply:
Its logic is merged into the `test_upgrade_script_s3_rbac_ok` except that I replaced the DF with a `write_pickle`.